src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ConfigurationTests.cs

```diff
@@ -0,0 +1,268 @@
+            RunConfigurationTest(redact, "Authentication.json", ConfigurationTestingMode.Authentication, expectedConfiguration);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void DefaultProcessTest(bool redact)
+        {
+            string expectedConfiguration = ConstructExpectedOutput(defaultProcessExpected);
+            RunConfigurationTest(redact, "DefaultProcess.json", ConfigurationTestingMode.DefaultProcess, expectedConfiguration);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void StorageTest(bool redact)
+        {
+            string expectedConfiguration = ConstructExpectedOutput(storageExpected);
+            RunConfigurationTest(redact, "Storage.json", ConfigurationTestingMode.Storage, expectedConfiguration);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void UrlsTest(bool redact)
+        {
+            string expectedConfiguration = ConstructExpectedOutput(urlsExpected);
+            RunConfigurationTest(redact, "URLs.json", ConfigurationTestingMode.URLs, expectedConfiguration);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void EgressTest(bool redact)
+        {
+            string expectedConfiguration = ConstructExpectedOutput(redact ? egressRedactedExpected : egressFullExpected);
+            RunConfigurationTest(redact, "Egress.json", ConfigurationTestingMode.Egress, expectedConfiguration);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void LoggingTest(bool redact)
+        {
+            string expectedConfiguration = ConstructExpectedOutput(loggingExpected);
+            RunConfigurationTest(redact, "Logging.json", ConfigurationTestingMode.Logging, expectedConfiguration);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void DiagnosticPortTest(bool redact)
+        {
+            string expectedConfiguration = ConstructExpectedOutput(diagnosticPortExpected);
+            RunConfigurationTest(redact, "DiagnosticPort.json", ConfigurationTestingMode.DiagnosticPort, expectedConfiguration);
+        }
+
+        private void CompareOutput(string actual, string expected)
+        {
+            Assert.Equal(CleanOutput(expected), CleanOutput(actual));
+        }
+
+        private string ConstructExpectedOutput(Dictionary<string, string> categoryMapping)
+        {
+            string expectedOutput = "{";
+
+            foreach (var key in orderedConfigurationKeys)
+            {
+                expectedOutput += "\"" + key + "\"" + ":";
+
+                if (categoryMapping.ContainsKey(key))
+                {
+                    string expectedPath = Path.Combine(Directory.GetCurrentDirectory(), "ExpectedConfigurations", categoryMapping[key]);
```

Comment:
I'm not sure current directory advisable. Is this value the same in the context of command line, visual studio, and ci tests? It's usually safer to use test settings (assuming xunit has them) or Assembly.GetExecutingAssembly().Location or something simlilar.

Reply:
Currently leaving as unresolved - I didn't look too much into this yet, but swapping in Assembly.GetExecutingAssembly().Location didn't work.

Reply:
Update: using `Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)` did work. The location was giving me the DLL, which naturally wasn't working for referencing relative files